assets/javascripts/birdeck.js

```diff
@@ -17,32 +22,34 @@ function renderPost(post) {
     + "</p>"
     + "<button id='delete-post' name='button-fetch' class='btn btn-default btn-xs'>Delete</button>"
     + "</div>"
-    )
+  )
+}
+
+function renderPosts(posts) {
+  return posts.map(renderPost)
+}
+
+function addPostToPage(post) {
+  $latestPosts.append(post);
+}
+
+function addPostsToPage(posts) {
+  $latestPosts.html(posts)
+}
+
+function handleError(xhr) {
+  console.log(xhr.responseText)
 }
 
 function fetchPostsButton() {
-  $("#button-fetch-posts").on("click", function(){
-    fetchPosts()
-    })
+  $("#button-fetch-posts").on("click", fetchPosts)
 }
 
 function fetchPosts() {
-  var newestItemID = parseInt($(".post").last().attr("data-id"))
-
-  $.ajax({
-    type:    "GET",
-    url:     "https://turing-birdie.herokuapp.com/api/v1/posts.json",
-    success: function(posts) {
-      $.each(posts, function(index, post) {
-        if (isNaN(newestItemID) || post.id > newestItemID) {
-          renderPost(post)
-        }
-      })
-    },
-    error: function(xhr) {
-      console.log(xhr.responseText)
-    }
-  })
+  $.getJSON("https://turing-birdie.herokuapp.com/api/v1/posts.json")
```

Comment:
They do now. :angel: I figure Module 3 doesn't really get into callbacks either and that's not even the preferred callback pattern. If they were going to use callbacks, then they should at least use the traditional pattern:

```js
$.getJSON("https://turing-birdie.herokuapp.com/api/v1/posts.json", function (posts) {
  addPostsToPage(posts.map(renderPost));
}, handleError);
```

IMHO opinion, that's way more confusing than just chaining promises.